### PR TITLE
Build model update

### PR DESCRIPTION
**Pull request by BryanRumsey**

### CODE REVIEW

**Review by mdip226 — COMMENTED** [review]
(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/CLE_examples_convergence.ipynb)
unused:
import numpy
parameter_values=None

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/Events.ipynb)
unused:
parameter_values=None

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/KineticModelForStyrenePolymerization.ipynb)
TimeSpan class not defined throws error

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/Live Output with Python Solvers.ipynb)
unused:
parameter_values=None 
typo:
"Resume Resume"

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/Model_Pre_Compilation.ipynb)
why perf counter say Time to complete 20 runs using pre compiled model and solver: 0.1910390830016695
when it took 10.3 sec
non-precompiled took 3m 19s and says 199.717

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/Negative_Timespan_Example.ipynb)
typos:
'we must swift'
***
## Process Results
***
To visualize the results with the infection event at `t=0` we must swift the time sub array in the results by the added simulation time.

#### Simulate Infection with past population data
To model population data from prior to the infection we need to modify the infect event trigger by the ent time of the add period.  We do this using the variable solver to modify the infect_start parameter.

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/Run GillesPy2 Simulations in Parallel.ipynb)
dask import, maybe add a comment about installation?

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/SBML_Import_Test.ipynb)
libsbml required import?

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/hybrid_continuous_species.ipynb)
(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/hybrid_switching_example.ipynb)
match case in filename

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/AdvancedFeatures/hybrid_switching_example.ipynb)
typo: threshold
***
## Run the Simulations
***
The hybrid switching condition is based on a coefficient of variance for each species.  The error threshhold for this condition can be user modified by instantiating the solver, and calling the run function from it, rather than the solver.  
  
The default value is `0.03` and is a threshold to the coefficient of variance for each species at the selected forward step (t + tau).

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/DataVisualization.ipynb)
typo: add ()
***
## Run the Simulations
***
model.run creates a Results object with a list of Trajectory objects.

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/Models/ExtraModels/Dimerization.ipynb)
unused named argument
cell 6 "run" can be model.run()

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/Models/ExtraModels/Opioid_Model.ipynb)
typo: opoid to opioid

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/Models/ExtraModels/Oregonator.ipynb)
oregonator drop off at t=3.5?

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/Models/ExtraModels/SpeedComparisons/hybrid_switching_decay_model_performance.ipynb)
unused math, random
match case in filename
unused named argument

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/Models/ExtraModels/Brusselator.ipynb)
unused named argument (i think in all of the ExtraModels/StartingModels, and the ones in the examples dir)


(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/examples/StartingModels/VilarOscillator/VilarOscillator.py)
maybe put this in another directory?, it's lonely in this one

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/gillespy2/solvers/cpp/example_models.py)
can't tell if this is used anywhere

(https://github.com/StochSS/GillesPy2/blob/bb8eabc583a570bbd4097705d8e4ea9ae453ed3d/test/TestingBattery.ipynb)
invalid syntax in cell 20 with the url
some unused model imports
that status bar is awesome

**Comment by mdip226** [comment]
The tests are still running but I might have to pause them for now and come back to them

File: test/test_python_solver_perf.py [hunk 1]
```diff
@@ -37,13 +37,13 @@ class TestPythonSolverPerf(TestCase):
...
             ],
             ODESolver: [
-                Oregonator()
+                build_oregonator()
             ],
             TauLeapingSolver: [
-                VilarOscillator()
+                build_vilar_oscillator()
             ],
         }
```
Comment (mdip226) [inline test/test_python_solver_perf.py:49]: `./test_python_solver_perf.py::TestPythonSolverPerf::test_python_solver_perf Failed with Error: A model is required to run the simulation.
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_python_solver_perf.py", line 57, in test_python_solver_perf
    perf_data = python_profiler.run_profiler(model, solver)
  File "/home/mdip/Projects/GillesPy2/test/profiling/python_profiler.py", line 43, in run_profiler
    solver.run(number_of_trajectories=trajectories, t=timesteps, timeout=0)
  File "/home/mdip/Projects/GillesPy2/gillespy2/solvers/numpy/ssa_solver.py", line 90, in run
    self = NumPySSASolver(model=model)
  File "/home/mdip/Projects/GillesPy2/gillespy2/solvers/numpy/ssa_solver.py", line 38, in __init__
    raise SimulationError("A model is required to run the simulation.")
gillespy2.core.gillespyError.SimulationError: A model is required to run the simulation.`

Comment (mdip226) [reply test/test_python_solver_perf.py:49]:
```
./test_python_solver_perf.py::TestPythonSolverPerf::test_python_solver_perf Failed with Error: A model is required to run the simulation.
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_python_solver_perf.py", line 57, in test_python_solver_perf
    perf_data = python_profiler.run_profiler(model, solver)
  File "/home/mdip/Projects/GillesPy2/test/profiling/python_profiler.py", line 43, in run_profiler
    solver.run(number_of_trajectories=trajectories, t=timesteps, timeout=0)
  File "/home/mdip/Projects/GillesPy2/gillespy2/solvers/numpy/ssa_solver.py", line 90, in run
    self = NumPySSASolver(model=model)
  File "/home/mdip/Projects/GillesPy2/gillespy2/solvers/numpy/ssa_solver.py", line 38, in __init__
    raise SimulationError("A model is required to run the simulation.")
gillespy2.core.gillespyError.SimulationError: A model is required to run the simulation.
```

File: test/test_hybrid_c_events.py [hunk 2]
```diff
@@ -4,26 +4,26 @@
...
+        
+        rate = gillespy2.Parameter(name="k1", expression=rate)
+        model.add_parameter(rate)
+
+        r1 = gillespy2.Reaction(
+            name="r1", reactants={s1: 1}, products={s2: 1}, rate=rate,
+        )
+        model.add_reaction(r1)
 
     def test_event_with_time_trigger(self):
```
Comment (mdip226) [inline test/test_hybrid_c_events.py:22]:
```
./test_hybrid_c_events.py::EventFeatures::test_event_with_time_trigger Failed with Error: 'NoneType' object has no attribute 'get_species'
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_hybrid_c_events.py", line 25, in test_event_with_time_trigger
    gillespy2.EventAssignment(variable=model.get_species('S1'), expression="100.0"),
AttributeError: 'NoneType' object has no attribute 'get_species'
```

File: test/test_hybrid_c_events.py [hunk 3]
```diff
@@ -36,10 +36,10 @@ def test_event_with_time_trigger(self):
         self.assertAlmostEqual(s1 + s2, 100.0, places=1)
 
     def test_event_with_species_trigger(self):
```
Comment (mdip226) [inline test/test_hybrid_c_events.py:38]:
```
./test_hybrid_c_events.py::EventFeatures::test_event_with_species_trigger Failed with Error: 'NoneType' object has no attribute 'get_species'
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_hybrid_c_events.py", line 41, in test_event_with_species_trigger
    gillespy2.EventAssignment(variable=model.get_species('S1'), expression="100.0"),
AttributeError: 'NoneType' object has no attribute 'get_species'
```

File: test/test_hybrid_c_events.py [hunk 4]
```diff
@@ -52,16 +52,16 @@ def test_event_with_species_trigger(self):
         self.assertFalse(np.any(result["S1"] <= 90.0), "Expected S1 > 90 for entire simulation")
 
     def test_delay_trigger_persistent(self):
-        model = EventFeatures.BaseEventModel(s1=100, s2=0, rate=1.0)
+        model = EventFeatures.build_base_event_model(s1=100, s2=0, rate=1.0)
         event1 = gillespy2.Event(name="ev1", assignments=[
-            gillespy2.EventAssignment(variable=model.s1, expression="0"),
-            gillespy2.EventAssignment(variable=model.s2, expression="0"),
-            gillespy2.EventAssignment(variable=model.rate, expression="0.0")
+            gillespy2.EventAssignment(variable=model.get_species('S1'), expression="0"),
```
Comment (mdip226) [inline test/test_hybrid_c_events.py:57]: these ones too
 | 
V

File: test/test_hybrid_c_events.py [hunk 5]
```diff
@@ -78,13 +78,13 @@ def test_delay_trigger_persistent(self):
         self.assertEqual(s2, 200, "Persistent event failed to fire")
 
     def test_trigger_priorities(self):
```
Comment (mdip226) [inline test/test_hybrid_c_events.py:80]: some in here

File: test/test_hybrid_c_events.py [hunk 6]
```diff
@@ -98,10 +98,10 @@ def test_trigger_priorities(self):
         self.assertEqual(s2, 100, "Events fired in an incorrect order")
 
     def test_use_values_from_trigger_time(self):
```
Comment (mdip226) [inline test/test_hybrid_c_events.py:100]: same

File: test/test_hybrid_c_events.py [hunk 7]
```diff
@@ -112,10 +112,10 @@ def test_use_values_from_trigger_time(self):
         self.assertGreater(s2, s1, "Event assignment did not assign values from trigger time")
 
     def test_initial_values(self):
-        model = EventFeatures.BaseEventModel(s1=0, s2=100.0, rate=1.0)
+        model = EventFeatures.build_base_event_model(s1=0, s2=100.0, rate=1.0)
 
         event = gillespy2.Event(name="ev1", assignments=[
-            gillespy2.EventAssignment(variable=model.s1, expression="S2/2"),
+            gillespy2.EventAssignment(variable=model.get_species('S1'), expression="S2/2"),
```
Comment (mdip226) [inline test/test_hybrid_c_events.py:118]: same

File: test/test_c_decode.py [hunk 8]
```diff
@@ -3,11 +3,11 @@
...
-from example_models import Degradation
+from example_models import build_degradation
 from gillespy2.solvers.cpp.c_decoder import BasicSimDecoder, IterativeSimDecoder
 
 class TestCDecode(unittest.TestCase):
-    model = Degradation()
+    model = build_degradation()
     solvers = [
         # Solver     List of trajectory numbers to try
         (ODECSolver,        [1]),
```
Comment (mdip226) [inline test/test_c_decode.py:13]: line 32, about the same for each solver:
```
test_c_decode.TestCDecode.test_run_output [Processing simulation output for each solver with different trajectory sizes] (number_of_trajectories=3, solver=<gillespy2.solvers.cpp.ssa_c_solver.SSACSolver object at 0x7f2e72086df0>) Failed: 'SSACSolver' object is not callable
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 582, in subTest
    yield
  File "/home/mdip/Projects/GillesPy2/test/test_c_decode.py", line 32, in test_run_output
    solver = solver(model=self.model)
TypeError: 'SSACSolver' object is not callable
```

File: test/test_basic_tau_leaping_solver.py [hunk 9]
```diff
@@ -16,16 +16,16 @@
...
 
 class TestBasicTauLeapingSolver(unittest.TestCase):
 
     def setUp(self):
-        self.model = Example()
-        self.model_no_tspan = ExampleNoTspan()
+        self.model = build_example()
+        self.model_no_tspan = build_example_no_tspan()
         self.solver = TauLeapingSolver(model=self.model)
         self.solver_no_tspan = TauLeapingSolver(model=self.model_no_tspan)
```
Comment (mdip226) [inline test/test_basic_tau_leaping_solver.py:31]:
```
./test_basic_tau_leaping_solver.py::TestBasicTauLeapingSolver::test_return_type Failed: 
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_basic_tau_leaping_solver.py", line 35, in test_return_type
    assert(isinstance(results, np.ndarray))
AssertionError
```

Comment (mdip226) [reply test/test_basic_tau_leaping_solver.py:31]:
```
./test_basic_tau_leaping_solver.py::TestBasicTauLeapingSolver::test_return_type_show_labels Failed with Error: isinstance() arg 2 must be a type or tuple of types
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_basic_tau_leaping_solver.py", line 43, in test_return_type_show_labels
    assert(isinstance(labels_results, results))
TypeError: isinstance() arg 2 must be a type or tuple of types
```

**Review by mdip226 — CHANGES_REQUESTED** [review]
one more:
```
./test_StochML.py::TestStochML::test_StochML_from_and_to_model Failed with Error: could not convert string to float: 'None'
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib/python3.8/unittest/case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "/usr/lib/python3.8/unittest/case.py", line 633, in _callTestMethod
    method()
  File "/home/mdip/Projects/GillesPy2/test/test_StochML.py", line 30, in test_StochML_from_and_to_model
    stochml_model = stochml.to_model('model')
  File "/home/mdip/Projects/GillesPy2/gillespy2/core/model.py", line 1331, in to_model
    model.volume = float(expr)
ValueError: could not convert string to float: 'None'
```

**Comment by mdip226** [comment]
i think that's it!

**Review by seanebum — APPROVED** [review]

**Review by briandrawert — APPROVED** [review]

**Pull request merged**